packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts

```diff
@@ -198,432 +164,500 @@ describe('quantic-generated-answer', () => {
+        });
 
-      beforeEach(() => {
-        mockSearchWithGeneratedAnswer(streamId);
-        mockStreamResponse(streamId, testErrorPayload);
-        visitGeneratedAnswer();
+        it('should log the generated answer stream end event', () => {
+          Expect.logGeneratedAnswerStreamEnd(streamId);
+        });
       });
 
-      it('should not display the component', () => {
-        Expect.displayGeneratedAnswerCard(false);
-      });
-    });
+      describe('when an error event is received', () => {
+        const streamId = crypto.randomUUID();
 
-    describe('when the stream connection fails', () => {
-      const streamId = crypto.randomUUID();
+        const testErrorPayload = {
+          finishReason: 'ERROR',
+          errorMessage: 'An error message',
+          errorCode: 500,
+        };
 
-      describe('Non-retryable error (4XX)', () => {
         beforeEach(() => {
-          mockSearchWithGeneratedAnswer(streamId);
-          mockStreamError(streamId, 406);
-          visitGeneratedAnswer();
-          cy.wait(getStreamInterceptAlias(streamId));
+          mockSearchWithGeneratedAnswer(streamId, param.useCase);
+          mockStreamResponse(streamId, testErrorPayload);
+          visitGeneratedAnswer({useCase: param.useCase});
         });
 
-        it('should not show the component', () => {
+        it('should not display the component', () => {
           Expect.displayGeneratedAnswerCard(false);
         });
       });
 
-      describe('Retryable error', () => {
-        [500, 429].forEach((errorCode) => {
-          describe(`${errorCode} error`, () => {
-            beforeEach(() => {
-              mockSearchWithGeneratedAnswer(streamId);
-              mockStreamError(streamId, errorCode);
-              visitGeneratedAnswer();
-            });
+      describe('when the stream connection fails', () => {
+        const streamId = crypto.randomUUID();
 
-            it('should retry the stream 3 times then offer a retry button', () => {
-              for (let times = 0; times < 3; times++) {
-                Expect.displayGeneratedAnswerCard(false);
-                cy.wait(getStreamInterceptAlias(streamId));
-              }
-              Expect.displayGeneratedAnswerCard(true);
+        describe('Non-retryable error (4XX)', () => {
+          beforeEach(() => {
+            mockSearchWithGeneratedAnswer(streamId, param.useCase);
+            mockStreamError(streamId, 406);
+            visitGeneratedAnswer({useCase: param.useCase});
+            cy.wait(getStreamInterceptAlias(streamId));
+          });
 
-              Actions.clickRetry();
-              cy.wait(InterceptAliases.Search);
-              Expect.logRetryGeneratedAnswer(streamId);
+          it('should not show the component', () => {
+            Expect.displayGeneratedAnswerCard(false);
+          });
+        });
+
+        describe('Retryable error', () => {
+          [500, 429].forEach((errorCode) => {
```

Comment:
are those the only 2 possible error codes we can receive?

Reply:
Yep.